python/cuml/linear_model/linear_regression.pyx

```diff
@@ -218,26 +219,38 @@ class LinearRegression(Base):
             'eig': 1
         }[algorithm]
 
-    def fit(self, X, y):
+    def fit(self, X, y, convert_dtype=False):
         """
         Fit the model with X and y.
 
         Parameters
         ----------
-        X : cuDF DataFrame
-            Dense matrix (floats or doubles) of shape (n_samples, n_features)
+        X : array-like (device or host) shape = (n_samples, n_features)
+            Dense matrix (floats or doubles) of shape (n_samples, n_features).
+            Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
+            ndarray, cuda array interface compliant array like CuPy
 
-        y: cuDF DataFrame
-           Dense vector (floats or doubles) of shape (n_samples, 1)
+        y : array-like (device or host) shape = (n_samples, 1)
+            Dense vector (floats or doubles) of shape (n_samples, 1).
+            Acceptable formats: cuDF Series, NumPy ndarray, Numba device
+            ndarray, cuda array interface compliant array like CuPy
+
+        convert_dtype : bool (default = False)
```

Comment:
Nicely documented, but maybe it's a little arbitrary that we make y into the dtype of X instead of the other way around?

Reply:
this is CPU numpy array for C++, so we don't want to transfer to GPU

Reply:
Decision was for memory usage, since X will be bigger always if we convert it instead it it'll hurt us in performance more than if we convert y